shipyard_airflow/plugins/rest_api_plugin.py

```diff
@@ -0,0 +1,744 @@
+airflow_version = airflow.__version__
+
+# Getting configurations from airflow.cfg file
+airflow_webserver_base_url = configuration.get('webserver', 'BASE_URL')
+airflow_base_log_folder = configuration.get('core', 'BASE_LOG_FOLDER')
+airflow_dags_folder = configuration.get('core', 'DAGS_FOLDER')
+
+# Using UTF-8 Encoding so that response messages don't have any characters in them that can't be handled
+os.environ['PYTHONIOENCODING'] = 'utf-8'
+
+
+"""
+Metadata that defines a single API:
+{
+    "name": "{string}",                     # Name of the API (cli command to be executed)
+    "description": "{string}",              # Description of the API
+    "airflow_version": "{string}",          # Version the API was available in to allow people to better determine if the API is available. (to be displayed on the Admin page)
+    "http_method": "{string}",              # HTTP method to use when calling the function. (Default: GET) (Optional)
+    "background_mode": {boolean},           # Whether to run the process in the background if its a CLI API (Optional)
+    "arguments": [                          # List of arguments that can be provided to the API
+        {
+            "name": "{string}",             # Name of the argument
+            "description": "{string}",      # Description of the argument
+            "form_input_type": "{string}",  # Type of input to use on the Admin page for the argument
+            "required": {boolean},          # Whether the argument is required upon submission
+            "cli_end_position": {int}       # In the case with a CLI command that the arguments value should be appended on to the end (for example: airflow trigger_dag some_dag_id), this is the position that the argument should be provided in the CLI command. (Optional)
+        }
+    ],
+    "fixed_arguments": [                    # List of arguments that will always be used by the API endpoint and can't be changed
+        {
+            "name": "{string}",             # Name of the argument
+            "description": "{string}",      # Description of the argument
+            "fixed_value": "{string}"       # Fixed value that will always be used
+        }
+    ],
+    "post_arguments": [                     # List of arguments that can be provided in the POST body to the API
+        {
+            "name": "{string}",             # Name of the argument
+            "description": "{string}",      # Description of the argument
+            "form_input_type": "{string}",  # Type of input to use on the Admin page for the argument
+            "required": {boolean},          # Whether the argument is required upon submission
+        }
+    ]
+},
+"""
+
+# Metadata about the APIs and how to call them. Representing them like this allows us to dynamically generate the APIs
+# in the admin page and dynamically execute them. This also allows us to easily add new ones.
+# API Object definition is described in the comment block above.
+apis_metadata = [
```

Comment:
Actually I am re-using the codes and have only removed certain parts of the codes that I felt was not relevant for our case.  I am only interested in the results of stdout and stderr in most cases (this piece of codes from GitHub was meant to wrap the Airflow CLI that has much more functionalities than the current experimental Airflow REST API).